src/HotChocolate/MongoDb/src/Data/Paging/AggregateFluentPagingContainer.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using HotChocolate.Types.Pagination;
+using MongoDB.Driver;
+
+namespace HotChocolate.Data.MongoDb.Paging
+{
+    internal class AggregateFluentPagingContainer<TEntity> : IMongoPagingContainer<TEntity>
+    {
+        public readonly IAggregateFluent<TEntity> _source;
+
+        public AggregateFluentPagingContainer(IAggregateFluent<TEntity> source)
+        {
+            _source = source;
+        }
+
+        public async Task<int> CountAsync(CancellationToken cancellationToken)
+        {
+            AggregateCountResult result = await _source
+                .Count()
+                .FirstOrDefaultAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            return (int)result.Count;
```

Comment:
result is null if count = 0. Currently exception instead of return 0;

`return (int)(result?.Count ?? 0);`

Reply:
fixed it